### ISSUE TITLE
[BUG] [AMQP] Websocket upgrade request's hostname must be the same as the HTTP connection host

### ISSUE THREAD

**Issue by anuchandy** [Azure/azure-sdk-for-java#31825]
Today we use the Event Hubs (or Service Bus) FQDN as the "hostname" + "host header" for the HTTP GET upgrade request to switch the protocol to WebSocket on an existing HTTP connection.

Such an upgrade request works if the existing HTTP connection had established to the FQDN of the Event Hubs.

On the other hand, if the HTTP connection had established to the Custom Endpoint (front-ending Event Hubs or Service Bus), then an upgrade request using the FQDN of the Event Hubs as the "hostname" + "host-header" may fail. Instead, the "hostname" + "host-header" should be the custom endpoint host.

For example, the server addressed by the Custom Endpoint may validate the "host-header" resulting 404 responses for the Upgrade request when the client expects a 101 response. Interestingly, not all custom endpoint host does the "host-header" validation; that's why we didn't encounter this before. For example, using App Gateway as a custom endpoint with multi-site setup kick off this validation; with basic-site setup, this validation is off.

In a nutshell, the Websocket upgrade request's hostname must be the same as the HTTP connection host.
